src/NativeScript/Metadata/Metadata.mm

```diff
@@ -207,8 +207,10 @@ int compareIdentifiers(const char* nullTerminated, const char* notNullTerminated
     } while (this->_topLevelIndex < this->_globalTable->buckets.count);
 }
 
+extern char startOfMetadataSection __asm("section$start$__DATA$__TNSMetadata");
```

Comment:
1. I tested it with CLI project, so it works when the runtime is dynamically linked into the application. 
2. I had problems with Mach-O APIs. I didn't manage to retrieve a pointer to the Mach-O __DATA section. The documented way to achive it just didn't work, it returns me a wrong pointer pointing somewhere else but not to the section I am looking for. This is what i tried: 
```
const struct section_64* metadataSection = getsectbyname("__DATA", "__TNSMetadata");
static MetaFile* instance(reinterpret_cast<MetaFile*>(metadataSection->addr)); // note: the cast is working correctly, trust me :)
```
I will spend more time on this, maybe I miss something.